src/tools/timemory-connector/README.md

```diff
@@ -0,0 +1,404 @@
+| `user_clock`   | CPU time spent in user-mode                              |
+| `cpu_util`     | CPU utilization                                          |
+| `page_rss`     | High-water (allocated) memory mark in region             |
+| `peak_rss`     | High-water (used) memory mark in region (excluding swap) |
+
+### Appending components
+
+Other components in addition to those specified by `TIMEMORY_COMPONENTS`
+can be enabled at runtime via the `TIMEMORY_COMPONENT_LIST_INIT` environment variable.
+These components may include:
+
+| Metric                  | Description                                                                        |
+| ----------------------- | ---------------------------------------------------------------------------------- |
+| `caliper`               | Enable [Caliper toolkit](https://github.com/LLNL/Caliper)                          |
+| `papi_array`            | Enable PAPI hardware counters                                                      |
+| `cpu_roofline_sp_flops` | Enable single-precision floating-point roofline on CPU                             |
+| `cpu_roofline_dp_flops` | Enable single-precision floating-point roofline on CPU                             |
+| `gperf_cpu_profiler`    | Enable gperftools CPU profiler (sampler)                                           |
+| `gperf_heap_profiler`   | Enable gperftools heap profiler                                                    |
+| `cuda_event`            | Enable `cudaEvent_t` for approximate kernel runtimes                               |
+| `nvtx_marker`           | Enable NVTX markers                                                                |
+| `cupti_activity`        | Enable CUPTI runtime tracing on NVIDIA GPUs                                        |
+| `cupti_counters`        | Enable CUPTI hardware counters on NVIDIA GPUs                                      |
+| `gpu_roofline_flops`    | Enable half-, single-, and double-precision floating-point roofline on NVIDIA GPUs |
+| `gpu_roofline_hp_flops` | Enable half-precision floating-point roofline on NVIDIA GPUs                       |
+| `gpu_roofline_sp_flops` | Enable single-precision floating-point roofline on NVIDIA GPUs                     |
+| `gpu_roofline_dp_flops` | Enable double-precision floating-point roofline on NVIDIA GPUs                     |
+
+Some components will conflict with each other and should not be enabled simulatenously. Such combinations include:
+
+- `papi_array` and `cpu_roofline` types
+- `cupti_activity` and `cupti_counters`
+- `cupti_activity`/`cupti_counters` and `gpu_roofline` types
+- Any hardware counters + `caliper` (when Caliper is also configured to collect hardware counters)
+
+```console
+export TIMEMORY_COMPONENT_LIST_INIT="cupti_activity,papi_array"
+```
+
+## Run kokkos application with PAPI recording enabled
+
+When timemory was installed with PAPI support, assuming `CMAKE_INSTALL_RPATH_USE_LINK_PATH=ON`, the timemory
```

Comment:
This is mildly different from above, up at line 55 I need to have the PAPI location always, down here there are CMake options which will sometimes handle this for me (though it looks from the CMakeLists like you force those on in the connector itself). Double check that this and line 55 line up how you want to and resolve this thread and I'll call that good

Reply:
Yea so I think that was probably just too much info. I default install with the rpath encoded now but I think I didn't at the time.

I actually think I'm going to replace using the library interface with a new runtime bundler that uses the template interface so there isn't an explicit link dependency and also allow users to write their own components and plug them in if they want to.